jax/lax/lax.py

```diff
@@ -1402,16 +1402,22 @@ def atan(x):
 @api.jit
 def sinh(x):
   r"""Elementwise hyperbolic sine: :math:`\mathrm{sinh}(x)`."""
+  dtype = _dtype(x)
```

Comment:
Maybe abstract this pattern (of up-casting for the computation then down-casting), since it's used on several unary-input unary-output ops? Could be a decorator.

Reply:
Done. Not sure if I prefer it this way but let's try it this way for a bit.